packages/@react-aria/combobox/docs/useComboBox.mdx

```diff
@@ -315,433 +323,184 @@ function Option({item, state, shouldUseVirtualFocus}) {
-  let {
-    popoverRef,
-    listBoxRef,
-    state,
-    shouldUseVirtualFocus,
-    ...otherProps
-  } = props;
+### Allow custom values
 
-  // Get props for the listbox.
-  // Prevent focus moving to listbox via shouldUseVirtualFocus
-  let {listBoxProps} = useListBox({
-    autoFocus: state.focusStrategy,
-    disallowEmptySelection: true,
-    shouldUseVirtualFocus,
-    ...otherProps
-  }, state, listBoxRef);
+By default, `useComboBoxState` doesn't allow users to specify a value that doesn't exist in the list of options and will revert the input value to
+the current selected value on blur. By specifying `allowsCustomValue`, this behavior is suppressed and the user is free to enter
+any value within the field.
 
-  // Handle events that should cause the popup to close,
-  // e.g. blur, clicking outside, or pressing the escape key.
-  let {overlayProps} = useOverlay({
-    onClose: () => state.close(),
-    shouldCloseOnBlur: true,
-    isOpen: state.isOpen,
-    isDismissable: true
-  }, popoverRef);
+```tsx example
+<ComboBox label="Favorite Animal" allowsCustomValue>
+  <Item key="red panda">Red Panda</Item>
+  <Item key="cat">Cat</Item>
+  <Item key="dog">Dog</Item>
+  <Item key="aardvark">Aardvark</Item>
+  <Item key="kangaroo">Kangaroo</Item>
+  <Item key="snake">Snake</Item>
+</ComboBox>
+```
 
-  // Add hidden <DismissButton> components at the end of the list
-  // to allow screen reader users to dismiss the popup easily.
-  return (
-    <div {...overlayProps} ref={popoverRef}>
-      <ul
-        {...mergeProps(listBoxProps, otherProps)}
-        ref={listBoxRef}
-        style={{
-          position: 'absolute',
-          width: '100%',
-          margin: '4px 0 0 0',
-          padding: 0,
-          listStyle: 'none',
-          border: '1px solid gray',
-          background: 'lightgray'
-        }}>
-        {[...state.collection].map(item => (
-          <Option
-            shouldUseVirtualFocus
-            key={item.key}
-            item={item}
-            state={state} />
-        ))}
-      </ul>
-      <DismissButton onDismiss={() => state.close()} />
-    </div>
-  )
-}
+### Custom filtering
 
-function Option({item, state, shouldUseVirtualFocus}) {
-  let ref = React.useRef();
-  let isDisabled = state.disabledKeys.has(item.key);
-  let isSelected = state.selectionManager.isSelected(item.key);
-  // Track focus via focusedKey state instead of with focus event listeners
-  // since focus never leaves the text input in a ComboBox
-  let isFocused = state.selectionManager.focusedKey === item.key;
+By default, `useComboBoxState` uses the filter function passed to the `defaultFilter` prop (in the above example, a
+"contains" function from `useFilter`). The filter function can be overridden by users of the `ComboBox` component by
+using the `items` prop to control the filtered list. When `items` is provided rather than `defaultItems`, the `useComboBoxState`
+does no filtering of its own.
```

Comment:
```suggestion
using the `items` prop to control the filtered list. When `items` is provided rather than `defaultItems`, `useComboBoxState`
does no filtering of its own.
```